### DISCUSSION TITLE
Inspired by Fabric -> PowerShell Weave

### DISCUSSION THREAD

**Discussion by dfinke** [danielmiessler/Fabric#71] [Show and tell]
Hello @danielmiessler. I'm [Doug Finke](https://github.com/dfinke) 15x Microsoft MVP. This project is great. I created the [PowerShell AI Assistant](https://github.com/dfinke/PowerShellAIAssistant) when OpenAI released the new API in November 2023. It mirrors the Python SDK. I also ported an earlier Python SDK version from Jan 2023 [PowerShell AI](https://github.com/dfinke/PowerShellAI) supporting Text completion and the early Chat approach.

I saw you demo Fabric which inspired me to create [PSWeave](https://github.com/dfinke/PSWeave). I borrowed your "system prompts" and added more like: CompareAndContrast, DateBuddy, MsGraphWizard, PostToX, PowerShellMenuWizard, PsRegexWizard.

It leverages the object flow engine in PowerShell that enables piping from one "prompt" to the next, like yours.
It also uses the new AI Assistant OpenAI API primitives - Assistant, Threads, Messages and Runs. It also all runs in the OpenAI backend.

Would love to chat. Awesome work @danielmiessler! - Doug

**Comment by danielmiessler** [comment]
Let's definitely connect! And let's get those Patterns into Fabric too!

**Comment by dfinke** [comment]
What is the best way to connect? I can DM you X
